en/pl/pl_tuning.rst

```diff
@@ -0,0 +1,204 @@
+
+::
+    
+    sp_name                      owner           sp_type               is_deterministic    
+ ========================================================================================
+    'pl_csql_not_deterministic'  'DBA'           'FUNCTION'            'NO'                
+    'pl_csql_deterministic'      'DBA'           'FUNCTION'            'YES' 
+
+In the above example, the pl_csql_not_deterministic function does not use query caching in correlated subqueries because it is **NOT DETERMINISTIC**.
+On the other hand, the pl_csql_deterministic function uses query caching in correlated subqueries because it is specified with the **DETERMINISTIC** keyword, optimizing performance.
+
+.. code-block:: sql
+    
+    ;trace on
+    SELECT (SELECT pl_csql_not_deterministic (t1.col1) FROM dual) AS results FROM dummy_tbl t1;
+
+::
+
+      results
+ =============
+            2
+            3
+            2
+            3
+ 
+ === Auto Trace ===
+    ...
+    Trace Statistics:
+      SELECT (time: 4, fetch: 11, fetch_time: 0, ioread: 0)
+        FUNC (time: 4, fetch: 2, ioread: 0, calls: 4)
+        SCAN (table: dba.dummy_tbl), (heap time: 0, fetch: 5, ioread: 0, readrows: 4, rows: 4)
+        SUBQUERY (correlated)
+          SELECT (time: 4, fetch: 6, fetch_time: 0, ioread: 0)
+            FUNC (time: 4, fetch: 2, ioread: 0, calls: 4)
+            SCAN (table: dual), (heap time: 0, fetch: 4, ioread: 0, readrows: 4, rows: 4)
+
+The pl_csql_not_deterministic function does not cache the results of correlated subqueries because it is **NOT DETERMINISTIC**.
+
+.. code-block:: sql
+    
+    ;trace on
+    SELECT (SELECT pl_csql_deterministic (t1.col1) FROM dual) AS results FROM dummy_tbl t1;
+
+::
+
+      results
+ =============
+            2
+            3
+            2
+            3
+
+ === Auto Trace ===
+    ...
+    Trace Statistics:
+      SELECT (time: 2, fetch: 9, fetch_time: 0, ioread: 0)
+        FUNC (time: 2, fetch: 2, ioread: 0, calls: 2)
+        SCAN (table: dba.dummy_tbl), (heap time: 0, fetch: 5, ioread: 0, readrows: 4, rows: 4)
+        SUBQUERY (correlated)
+          SELECT (time: 2, fetch: 4, fetch_time: 0, ioread: 0)
+            FUNC (time: 2, fetch: 2, ioread: 0, calls: 2)
+            SCAN (table: dual), (heap time: 0, fetch: 2, ioread: 0, readrows: 2, rows: 2)
+            SUBQUERY_CACHE (hit: 2, miss: 2, size: 150808, status: enabled)
+
+In the trace results of the pl_csql_deterministic function, the **SUBQUERY_CACHE** item is displayed (hit: 2, miss: 2, size: 150808, status: enabled), and the number of records read (**readrows**) in the **SCAN (table: dual)** at the top is reduced compared to the **NOT DETERMINISTIC** example.
+
+.. warning::
+
+        * The **DETERMINISTIC** attribute is not supported in stored procedures.
+        * If the **DETERMINISTIC** attribute is used in internal implementations that return non-deterministic results, it may not return the expected results.
```

Comment:
```suggestion
        * If a function that returns non-deterministic results is used the **DETERMINISTIC** option, it may not return the expected results.
```